pkg/scheduler/nodeinfo/snapshot.go

```diff
@@ -20,12 +20,14 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
-// Snapshot is a snapshot of cache NodeInfo. The scheduler takes a
-// snapshot at the beginning of each scheduling cycle and uses it for its
-// operations in that cycle.
+// Snapshot is a snapshot of cache NodeInfo and NodeTree order. The scheduler takes a
+// snapshot at the beginning of each scheduling cycle and uses it for its operations in that cycle.
 type Snapshot struct {
+	// NodeInfoMap a map of node name to a snapshot of its NodeInfo.
 	NodeInfoMap map[string]*NodeInfo
-	Generation  int64
+	// NodeInfoList is the list of nodes as ordered in the cache's nodeTree.
+	NodeInfoList []*NodeInfo
```

Comment:
Can we make it `NodeNameList []string`, and then in each `checkNode()`, the "nodeInfo" can be fetched by: `nodeInfo := g.nodeInfoSnapshot. NodeInfoMap[NodeNameList[i]]` (or from a helper method)?

Reply:
sounds good.